### PR TITLE
Branch2

### PR DESCRIPTION
**Pull request by MartinPetrakiev**

### CODE REVIEW

File: Fill_the_matrix/Program.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        }
+
+        static void Main(string[] args)
+        {
+            int n = Convert.ToInt32(Console.ReadLine());
+            string type = Console.ReadLine()!;
+
+            int[,] matrix = new int[n, n];
+
+            if (type == "a")
```
Comment (mvlt) [inline Fill_the_matrix/Program.cs:122]: Better switch case when so much options

File: Fill_the_matrix/Program.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
+﻿using System;
+
+namespace MyProject
+{
+    class Program
+    {
+        static void FillPatternA(int[,] matrix, int n)
```
Comment (mvlt) [inline Fill_the_matrix/Program.cs:7]: No need of second param in all functions

File: Fill_the_matrix/Program.cs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                FillPatternB(matrix, n);
+            }
+            else if (type == "c")
+            {
+                FillPatternC(matrix, n);
+            }
+            else if (type == "d")
+            {
+                FillPatternD(matrix, n);
+            }
```
Comment (mvlt) [inline Fill_the_matrix/Program.cs:137]: What if you get "e"

**Review by mvlt — CHANGES_REQUESTED** [review]
refactoring needed and finish for D

**Review by mvlt — CHANGES_REQUESTED** [review]
No description

File: Maximal_sum/Program.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    static void Main(string[] args)
+    {
+        string[] dimensions = Console.ReadLine().Split();
+        int n = Convert.ToInt32(dimensions[0]);
+        int m = Convert.ToInt32(dimensions[1]);
+
+        int[,] matrix = InitializeInt2DArray(n, m);
+
+        CalculateMaxSumOf3x3(matrix, n, m);
```
Comment (mvlt) [inline Maximal_sum/Program.cs:48]: No need of last two params

File: Binary_search/Program.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+        Console.WriteLine();
+        Console.WriteLine("Largest number <= k in the array is:");
+
+        int indexK = Array.BinarySearch(array, k);
+
+        if(indexK < 0)
+        {
+            Console.WriteLine(array[~indexK - 1]);
```
Comment (mvlt) [inline Binary_search/Program.cs:35]: What will happen if ~indexK equals 0?

**Review by mvlt — CHANGES_REQUESTED** [review]
All scenarios should to be handled

**Review by mvlt — CHANGES_REQUESTED** [review]
Check what is doing the second parameter passed to Array.Sort build in function

File: Fill_the_matrix/Program.cs (hunk 1 above)
Comment (MartinPetrakiev) [reply Fill_the_matrix/Program.cs]: Fill_the_matrix solution was not published. This was the initial commit.
code refactoring of iunitial in commit
[76c6369](https://github.com/MartinPetrakiev/c_sharp_multidymentional_arrays/pull/2/commits/76c6369e528a67abebfc26807c6180a7974ef554)
Includes swtich case.

File: Binary_search/Program.cs (hunk 5 above)
Comment (MartinPetrakiev) [reply Binary_search/Program.cs]: scenario handled at commit
[32b0db4](https://github.com/MartinPetrakiev/c_sharp_multidymentional_arrays/pull/2/commits/32b0db467428387e267a685726b06f851306cb32)

**Pull request merged**